### PR TITLE
Fixed minor error while calculating span accuracy

### PR DESCRIPTION
**Pull request by abhishek0318**
Inputs to BooleanAccuracy must have the same dimensions. Here, the first argument has dimensions `(batch size, 2)`, while the second argument has dimensions `(batch size, 1, 2)`. Changing `torch.stack` to `torch.cat` makes the dimension of the second argument `(batch size, 2)`.

Note that span accuracy obtained earlier was not incorrect because of the way accuracy was being computed internally.

### CODE REVIEW

**Comment by kl2806** [comment]
Thanks! Would you mind raising a config error checking that they are the same dimension similar to: https://github.com/allenai/allennlp/blob/master/allennlp/training/metrics/categorical_accuracy.py#L48
and a test that checks for this similar to https://github.com/allenai/allennlp/blob/1938a5af6d1ea3a2d8dccbb831f82990800fa3e1/allennlp/tests/training/metrics/categorical_accuracy_test.py#L79

**Comment by matt-gardner** [comment]
Minor point: config errors should only be used when the error reasonably came from a configuration file problem. That's probably not the case here, and a value error would be more appropriate.

**Comment by abhishek0318** [comment]
> Thanks! Would you mind raising a config error checking that they are the same dimension similar to and a test that checks for this similar to.

Sure. I will do it over the weekend.

**Comment by abhishek0318** [comment]
I have added the checks and the corresponding tests. Please take a look.
